dlt/destinations/impl/redshift/redshift.py

```diff
@@ -86,6 +86,12 @@ def run(self) -> None:
                 f" 'aws_access_key_id={aws_access_key};aws_secret_access_key={aws_secret_key}'"
             )
 
+            aws_region = self._staging_credentials.region_name
+            if aws_region:
+                region = f"region '{aws_region}'"
+
+        if self._staging_iam_role:
```

Comment:
you changed the logic here so both IAM_ROLE and CREDENTIALS will be present. They are exclusive auth methods and in previous code IAM_ROLE took precedence. That must still be the case!

Reply:
why?
if IAM_ROLE is defined, `credentials` will be overwritten to iam-string, so iam still takes precedence (if, if clause)